src/download.rs

```diff
@@ -307,6 +307,63 @@ pub fn unpack_bins(dir: &Path, dst_dir: &Path) -> Result<Vec<PathBuf>> {
     Ok(downloaded)
 }
 
+/// Read the version (as a plain String) used by the `fuels` dependency, if it exists.
+fn fuels_version_from_toml(toml: toml_edit::Document) -> Result<String> {
+    if let Some(deps) = toml.get("dependencies") {
+        if let Some(fuels) = deps.get("fuels") {
+            let version = match fuels.as_value() {
+                Some(toml_edit::Value::String(s)) => s.value().to_string(),
+                Some(toml_edit::Value::InlineTable(t)) => t.get("version").map_or_else(
+                    || "".to_string(),
+                    |v| v.as_str().unwrap_or_default().to_string(),
+                ),
+                _ => String::default(),
+            };
+            println!("vs: {version}");
+
+            return Ok(version);
+        } else {
+            bail!("'fuels' dependency does not exist");
+        };
+    };
+
+    bail!("the table 'dependencies' does not exist");
+}
+
+/// Fetches the Cargo.toml of a component in its repository and tries to read the version of
+/// the `fuels` dependency.
+pub fn fetch_fuels_version(cfg: &DownloadCfg) -> Result<String> {
```

Comment:
It looks like we're fetching fuels versions upon `fuelup show` - is that right?

Rather than fetching the version manually from `fuelup`, I wonder if the `fuels` versions should be determined as an optional field as a part of the channel/component information generated by CI? Perhaps this might lower the number of fetches that needs to happen (and improve response time) when a user runs `fuelup show`?

Reply:
This is called by `cache_fuels_version()`, which in turn is called in `install_component()` ([link](https://github.com/FuelLabs/fuelup/pull/410/files/f4326257cc185b5ddab7096082b3bbfae2edaa1e#diff-41a991785a2eba02345db5fdcbf15b660f27b2841b8f61a084a7db66e69bf3eeR46-R60)). So `fuels` version is cached only on install.  So actually when `fuelup show` is called we're just reading from the cached version [here](https://github.com/FuelLabs/fuelup/pull/410/files/f4326257cc185b5ddab7096082b3bbfae2edaa1e#diff-adb596276059ced1ef81d909863ca19057da292efe1aa1d90288ad87df021626R143)!

> I wonder if the fuels versions should be determined as an optional field as a part of the channel/component information generated by CI?

I went into this in the PR text actually - my main reasoning for going for something like this instead is so we can get the `fuels` version on a component-level instead of a toolchain-level. The ugly way we have to hardcode the fetching logic is unavoidable and is just a case of if we want to hide it behind building the channels or use it in fuelup itself.

Reply:
Ahh sorry bing, I should have read more closely, thanks for clarifying! Doing so during installation sounds fine :+1: